Assets/Scripts/Models/WorldGenerator.cs

```diff
@@ -102,16 +102,17 @@ public static void Generate(World world, int seed)
 
                                         int weight = inv.StackSize; // In stacksize the weight was cached
                                         currentweight += weight;
-                                    if (inv.objectType == "Raw Iron" || inv.objectType == "Uranium")
-                                    {
-                                        Furniture mine = PrototypeManager.Furniture.GetPrototype("mine").Clone();
-                                        mine.Parameters["ore_type"].SetValue(inv.objectType.ToString());
-                                        world.PlaceFurniture(mine, t, false);                           
-                                        break;
-                                    }
 
                                         if (randomweight <= currentweight)
                                         {
+                                            if (inv.objectType == "Raw Iron" || inv.objectType == "Uranium")
+                                            {
+                                                Furniture mine = PrototypeManager.Furniture.GetPrototype("mine").Clone();
+                                                mine.Parameters["ore_type"].SetValue(inv.objectType.ToString());
+                                                world.PlaceFurniture(mine, t, false);
```

Comment:
Could you wrap this block in a check like this to remove the errors?

```
Furniture minePrototype = PrototypeManager.Furniture.GetPrototype("mine");
if (minePrototype.IsValidPosition(t))
{
    // place mine here.
}
```

There seems to be something off with the indentation in this block as well. My MonoDevelop was very unhappy about it. You don't have to fix that, but it would be a plus point if you did.